rosidl_runtime_py/rosidl_runtime_py/convert.py

```diff
@@ -0,0 +1,148 @@
+        else:
+            value = ''.join([chr(c) for c in value])
+    elif isinstance(value, str):
+        if truncate_length is not None and len(value) > truncate_length:
+            value = value[:truncate_length] + '...'
+    elif isinstance(value, tuple) or isinstance(value, list):
+        if truncate_length is not None and len(value) > truncate_length:
+            # Truncate the sequence
+            value = value[:truncate_length]
+            # Truncate every item in the sequence
+            value = type(value)([_convert_value(v, truncate_length) for v in value] + ['...'])
+        else:
+            # Truncate every item in the list
+            value = type(value)([_convert_value(v, truncate_length) for v in value])
+    elif isinstance(value, dict) or isinstance(value, OrderedDict):
+        # Convert each key and value in the mapping
+        new_value = {} if isinstance(value, dict) else OrderedDict()
+        for k, v in value.items():
+            # Don't truncate keys because that could result in key collisions and data loss
+            new_value[_convert_value(k)] = _convert_value(v, truncate_length=truncate_length)
+        value = new_value
+    elif not any(isinstance(value, t) for t in (bool, float, int)):
+        # Assuming value is a message since it is neither a collection nor a primitive type
+        value = message_to_ordereddict(value, truncate_length=truncate_length)
+    return value
+
+
+def message_to_csv(msg: Any, truncate_length: int = None) -> str:
+    """
+    Convert a ROS message to string of comma-separated values.
+
+    :param msg: The ROS message to convert.
+    :param truncate_length: Truncate values for all message fields to this length.
+        This does not truncate the list of message fields.
+    :returns: A string of comma-separated values representing the input message.
+    """
+    def to_string(val):
+        nonlocal truncate_length
+        r = ''
+        if any(isinstance(val, t) for t in [list, tuple]):
+            for i, v in enumerate(val):
+                if r:
+                    r += ','
+                if truncate_length is not None and i >= truncate_length:
+                    r += '...'
+                    break
+                r += to_string(v)
+        elif any(isinstance(val, t) for t in [bool, bytes, float, int, str]):
+            if any(isinstance(val, t) for t in [bytes, str]):
+                if truncate_length is not None and len(val) > truncate_length:
+                    val = val[:truncate_length]
+                    if isinstance(val, bytes):
+                        val += b'...'
+                    else:
+                        val += '...'
+            r = str(val)
+        else:
+            r = message_to_csv(val, truncate_length)
+        return r
+    result = ''
+    # We rely on __slots__ retaining the order of the fields in the .msg file.
+    for field_name in msg.__slots__:
+        value = getattr(msg, field_name, None)
```

Comment:
Why should `. None` be passed? It seems like it is guaranteed that the `field_name` exists in the `msgs`?

Reply:
Not sure, this was from the original implementation. But I agree it looks like it's not necessary; removed in 4171c5a.